course/src/main/java/in/testpress/course/fragments/BaseContentListFragment.kt

```diff
@@ -0,0 +1,116 @@
+package `in`.testpress.course.fragments
+
+import `in`.testpress.course.R
+import `in`.testpress.course.TestpressCourse
+import `in`.testpress.databinding.BaseContentListLayoutBinding
+import `in`.testpress.fragments.EmptyViewFragment
+import `in`.testpress.fragments.EmptyViewListener
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ProgressBar
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
+import com.facebook.shimmer.ShimmerFrameLayout
+
+abstract class BaseContentListFragment: Fragment(), EmptyViewListener {
+
+    private lateinit var binding : BaseContentListLayoutBinding
+    private lateinit var loadingPlaceholder: ShimmerFrameLayout
+
+    // Protected properties are used in sub classes
+    protected lateinit var contentsURL: String
+    protected var chapterId: Long = -1
+    protected var productSlug: String? = null
+    protected var courseId: Long = -1
+    protected lateinit var swipeRefreshLayout: SwipeRefreshLayout
+    protected lateinit var recyclerView: RecyclerView
+    protected lateinit var progressBar: ProgressBar
+    protected lateinit var emptyViewFragment: EmptyViewFragment
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        parseArguments()
+        initializeViewModel()
+    }
```

Comment:
These should not be in base class. Let derived class override onCreate and handle these